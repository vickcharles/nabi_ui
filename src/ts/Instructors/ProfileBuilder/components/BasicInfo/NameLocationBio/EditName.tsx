import * as React from 'react';

import Button from 'material-ui/Button';
import { FormControl, FormControlLabel } from 'material-ui/Form';
import Radio, { RadioGroup } from 'material-ui/Radio';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog';

interface EditNameProps {
  firstName: string;
  lastName: string;
  isFormDialogOpen: boolean;
  closeHandler: () => void;
}
const EditName: React.StatelessComponent<EditNameProps> = props => {
  const lastNameinitial = props.lastName.charAt(0);

  return (
    <div>
      <Dialog
        open={props.isFormDialogOpen}
        onClose={props.closeHandler}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Edit Name</DialogTitle>
        <DialogContent>
          <DialogContentText className="nabi-margin-bottom-small">
            Choose display name
          </DialogContentText>
          <FormControl component="fieldset" required={true}>
          <RadioGroup
            name="displayName"
            value=""
          >
            <FormControlLabel 
              className="nabi-margin-bottom-xsmall"
              value="" 
              control={<Radio />} 
              label={`Show my full name: ${props.firstName} ${props.lastName}`} 
            />
            <FormControlLabel 
              value="" 
              control={<Radio />} 
              label={`Show only my first name and last initial: ${props.firstName} ${lastNameinitial}.`}
            />
          </RadioGroup>
        </FormControl>
        </DialogContent>
        <DialogActions>
          <Button onClick={props.closeHandler} color="default">
            Close
          </Button>
          <Button variant="raised" onClick={props.closeHandler} color="primary">
            Save
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default EditName;
