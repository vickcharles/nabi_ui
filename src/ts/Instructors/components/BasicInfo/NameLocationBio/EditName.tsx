import * as React from 'react';

import Button from 'material-ui/Button';
import { FormControl, FormControlLabel } from 'material-ui/Form';
import Radio, { RadioGroup } from 'material-ui/Radio';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog';

/**
 * Props for EditName
 * @interface {EditNameProps}
 */
interface EditNameProps {
  displayName: string;
  firstName: string;
  lastName: string;
  isFormDialogOpen: boolean;
  closeHandler: () => void;
  handleChange: (event: React.FormEvent<{}>) => void;
  handleSubmit: (event: React.FormEvent<{}>) => void;
}

/**
 * Displays a dialog to change the user's displayName
 * @param {EditNameProps} props - The component's props
 */
const EditName: React.StatelessComponent<EditNameProps> = props => {
  const lastNameinitial = props.lastName.charAt(0);
  const fullName = `${props.firstName} ${props.lastName}`;
  const nameAndLasInitial = `${props.firstName} ${lastNameinitial}.`;

  return (
    <div>
      <Dialog
        open={props.isFormDialogOpen}
        onClose={props.closeHandler}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Edit Name</DialogTitle>
        <DialogContent>
          <DialogContentText className="nabi-margin-bottom-small">
            Choose display name
          </DialogContentText>
          <FormControl component="fieldset" required={true}>
          <RadioGroup
            name="displayName"
            value={props.displayName}
            onChange={props.handleChange}
          >
            <FormControlLabel 
              className="nabi-margin-bottom-xsmall"
              value={fullName}
              control={<Radio />} 
              label={`Show my full name: ${fullName}`} 
            />
            <FormControlLabel 
              value={nameAndLasInitial}
              control={<Radio />} 
              label={`Show only my first name and last initial: ${nameAndLasInitial}`}
            />
          </RadioGroup>
        </FormControl>
        </DialogContent>
        <DialogActions>
          <Button onClick={props.closeHandler} color="default">
            Close
          </Button>
          <Button variant="raised" onClick={props.handleSubmit} color="primary">
            Save
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default EditName;
