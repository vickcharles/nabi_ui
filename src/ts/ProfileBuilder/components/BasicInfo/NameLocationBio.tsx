import * as React from 'react';
import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';
import Divider from 'material-ui/Divider';

interface NameLocationBioProps {
    firstName: string;
    lastName: string;
    zipCode: string;
    changeBio: (event: React.FormEvent<{}>) => void;
  }

const NameLocationBio: React.StatelessComponent<NameLocationBioProps> = props => {
  return (
    <div>
      <Typography className="nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall">
        {`${props.firstName} ${props.lastName}`}
      </Typography>
      
      <Typography className="nabi-text-center nabi-margin-bottom-medium">
        {props.zipCode}
      </Typography>

      <Typography className="nabi-margin-top-small" variant="body2">
        Bio *
      </Typography>

      <TextField
        id="bio"
        margin="normal"
        name="bio"
        onChange={props.changeBio}
        placeholder="What would you like your students to know  about you?"
        required={true}
        multiline={true}
        fullWidth={true}
        rows={6}
      />

      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default NameLocationBio;
