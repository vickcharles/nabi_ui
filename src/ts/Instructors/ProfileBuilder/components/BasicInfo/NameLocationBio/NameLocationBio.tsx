import * as React from 'react';

import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';
import Divider from 'material-ui/Divider';
import Edit from '@material-ui/icons/Edit';
import IconButton from 'material-ui/IconButton';
import EditName from './EditName';

interface NameLocationBioProps {
  firstName: string;
  lastName: string;
  zipCode: string;
  changeBio: (event: React.FormEvent<{}>) => void;
  blurBio: (event: React.FormEvent<{}>) => void;
}

interface NameLocationBioState {
  isModalOpenEditName: boolean;
  hovered: boolean;
}

export class NameLocationBio extends React.Component<NameLocationBioProps, NameLocationBioState> {
  constructor(props: NameLocationBioProps) {
    super(props);

    this.state = {
      isModalOpenEditName: false,
      hovered: false
    };
  }

  highlightedClass = () => {
    if (this.state.hovered) {
      return 'nabi-highlighted-text';
    } else {
      return '';
    }
  }

  onMouseOver = () => {
    this.setState({ hovered: true });
  }

  onMouseOut = () => {
    this.setState({ hovered: false });
  }

  public render(): JSX.Element {
    const closeEditName = () => this.setState({ isModalOpenEditName: false });

    return (
      <div>
        <div className="nabi-text-center"> 
          <Typography 
            className={`nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall 
            nabi-display-inline-block ${this.highlightedClass()}`}
          >
            {`${this.props.firstName} ${this.props.lastName}`}
          </Typography>
          <IconButton 
            onClick={() => this.setState({ isModalOpenEditName: true })}
            onMouseOver={this.onMouseOver} 
            onMouseOut={this.onMouseOut}
          >
            <Edit/>
          </IconButton>
        </div>
        
        <Typography className="nabi-text-center nabi-margin-bottom-medium">
          {this.props.zipCode}
        </Typography>

        <Typography className="nabi-margin-top-small" variant="body2">
          Bio *
        </Typography>

        <TextField
          id="bio"
          margin="normal"
          name="bio"
          onChange={this.props.changeBio}
          onBlur={this.props.blurBio}
          placeholder="What would you like your students to know  about you?"
          required={true}
          multiline={true}
          fullWidth={true}
          rows={6}
        />

        <Divider className="nabi-margin-top-small"/>

        <EditName 
          firstName={this.props.firstName}
          lastName={this.props.lastName}
          isFormDialogOpen={this.state.isModalOpenEditName}
          closeHandler={closeEditName}
        />
      </div>
    );
  }
}

export default NameLocationBio;
