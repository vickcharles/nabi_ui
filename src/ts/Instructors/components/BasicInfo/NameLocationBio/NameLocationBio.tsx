import * as React from 'react';

import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';
import Divider from 'material-ui/Divider';
import Edit from '@material-ui/icons/Edit';
import IconButton from 'material-ui/IconButton';

import EditName from './EditName';
import { UserState } from '../../../../Users/model';

interface NameLocationBioOwnProps {
  user: UserState;
  bioChange: (event: React.FormEvent<{}>) => void;
  bioBlur: (event: React.FormEvent<{}>) => void;
  displayName: string;
  updateName: (event: React.FormEvent<{}>) => void;
  handleChange: (event: React.FormEvent<{}>) => void;
}
/**
 * Props for NameLocationBio
 * @interface NameLocationBioProps
 */
interface NameLocationBioProps extends
NameLocationBioOwnProps { }

/**
 * State for NameLocationBio
 * @interface NameLocationBioState
 */
interface NameLocationBioState {
  isModalOpenEditName: boolean;
  hovered: boolean;
}

/**
 * Contains name, location and bio section in BasicInfo
 * @class NameLocationBio
 * @extends React.Component<NameLocationBioProps, NameLocationBioState>
 */
export class NameLocationBio extends React.Component<NameLocationBioProps, NameLocationBioState> {
  constructor(props: NameLocationBioProps) {
    super(props);

    this.state = {
      isModalOpenEditName: false,
      hovered: false
    };

    this.closeModal = this.closeModal.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  highlightedClass = () => {
    if (this.state.hovered) {
      return 'nabi-highlighted-text';
    } else {
      return '';
    }
  }

  onMouseOver = () => {
    this.setState({ hovered: true });
  }

  onMouseOut = () => {
    this.setState({ hovered: false });
  }

  public closeModal(): void {
    this.setState({ isModalOpenEditName: false });
  }

  public handleSubmit(event: any): void {
    this.props.updateName(event);
    this.closeModal();
  }

  public render(): JSX.Element {
    return (
      <div>
        <div className="nabi-text-center"> 
          <Typography 
            className={`nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall 
            nabi-display-inline-block ${this.highlightedClass()}`}
          >
            {`${this.props.displayName}`}
          </Typography>
         
          <IconButton 
            onClick={() => this.setState({ isModalOpenEditName: true })}
            onMouseOver={this.onMouseOver} 
            onMouseOut={this.onMouseOut}
          >
            <Edit/>
          </IconButton>
        </div>
        
        <Typography className="nabi-text-center">
        {`${this.props.user.city}, ${this.props.user.state} ${this.props.user.zipCode}`}
        </Typography>

        <Typography className="nabi-margin-top-small" variant="body2">
          Bio
        </Typography>

        <TextField
          id="bio"
          margin="normal"
          name="bio"
          onChange={this.props.bioChange}
          onBlur={this.props.bioBlur}
          placeholder="What would you like your students to know  about you?"
          required={true}
          multiline={true}
          fullWidth={true}
          rows={6}
        />

        <Divider className="nabi-margin-top-small"/>

        <EditName 
          firstName={this.props.user.firstName}
          lastName={this.props.user.lastName}
          isFormDialogOpen={this.state.isModalOpenEditName}
          closeHandler={this.closeModal}
          handleChange={this.props.handleChange}
          displayName={this.props.displayName}
          handleSubmit={this.handleSubmit}
        />
      </div>
    );
  }
}

export default NameLocationBio;
