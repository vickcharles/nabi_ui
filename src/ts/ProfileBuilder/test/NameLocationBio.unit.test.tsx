import * as React from 'react';
import NameLocationBio from '../components/BasicInfo/NameLocationBio';
import { shallow } from 'enzyme';

describe('NameLocationBio', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <NameLocationBio 
        firstName="foo"
        lastName="bar"
        zipCode="02920"
      />
    );
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
