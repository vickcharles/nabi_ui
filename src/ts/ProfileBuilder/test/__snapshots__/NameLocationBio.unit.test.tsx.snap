// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameLocationBio Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NameLocationBio
    changeBio={[MockFunction]}
    firstName="foo"
    lastName="bar"
    zipCode="02920"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          className="nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall"
        >
          foo bar
        </WithStyles(Typography)>,
        <WithStyles(Typography)
          className="nabi-text-center nabi-margin-bottom-medium"
        >
          02920
        </WithStyles(Typography)>,
        <WithStyles(Typography)
          className="nabi-margin-top-small"
          variant="body2"
        >
          Bio *
        </WithStyles(Typography)>,
        <TextField
          fullWidth={true}
          id="bio"
          margin="normal"
          multiline={true}
          name="bio"
          onChange={[MockFunction]}
          placeholder="What would you like your students to know  about you?"
          required={true}
          rows={6}
          select={false}
        />,
        <WithStyles(Divider)
          className="nabi-margin-top-small"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "foo bar",
          "className": "nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall",
        },
        "ref": null,
        "rendered": "foo bar",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "02920",
          "className": "nabi-text-center nabi-margin-bottom-medium",
        },
        "ref": null,
        "rendered": "02920",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Bio *",
          "className": "nabi-margin-top-small",
          "variant": "body2",
        },
        "ref": null,
        "rendered": "Bio *",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fullWidth": true,
          "id": "bio",
          "margin": "normal",
          "multiline": true,
          "name": "bio",
          "onChange": [MockFunction],
          "placeholder": "What would you like your students to know  about you?",
          "required": true,
          "rows": 6,
          "select": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "nabi-margin-top-small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            className="nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall"
          >
            foo bar
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className="nabi-text-center nabi-margin-bottom-medium"
          >
            02920
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className="nabi-margin-top-small"
            variant="body2"
          >
            Bio *
          </WithStyles(Typography)>,
          <TextField
            fullWidth={true}
            id="bio"
            margin="normal"
            multiline={true}
            name="bio"
            onChange={[MockFunction]}
            placeholder="What would you like your students to know  about you?"
            required={true}
            rows={6}
            select={false}
          />,
          <WithStyles(Divider)
            className="nabi-margin-top-small"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo bar",
            "className": "nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall",
          },
          "ref": null,
          "rendered": "foo bar",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "02920",
            "className": "nabi-text-center nabi-margin-bottom-medium",
          },
          "ref": null,
          "rendered": "02920",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Bio *",
            "className": "nabi-margin-top-small",
            "variant": "body2",
          },
          "ref": null,
          "rendered": "Bio *",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fullWidth": true,
            "id": "bio",
            "margin": "normal",
            "multiline": true,
            "name": "bio",
            "onChange": [MockFunction],
            "placeholder": "What would you like your students to know  about you?",
            "required": true,
            "rows": 6,
            "select": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "nabi-margin-top-small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
